low-level/framework/messaging/egress_accumulated_msgs_processor.py

```diff
@@ -122,7 +128,11 @@ def run(self):
                             continue
                     if "sensor_response_type" in dict_msg["message"]:
                         logger.info(f"Publishing Accumulated Alert: {message}")
-                    self._producer.send([message])
+                    if isinstance(self._producer, MessageProducer):
+                        self._producer.send([message])
```

Comment:
If this fails, what happens to current message being processed ?

Reply:
As discussed, changed get method in store_queue, and added separate delete method for deleting an element from the queue
Deleting element only if msg successfully sent - line134.

Reply:
Ok. This looks fine